storage/client.go

```diff
@@ -398,6 +398,15 @@ func (c Client) exec(verb, url string, headers map[string]string, body io.Reader
 		return nil, errors.New("azure/storage: error creating request: " + err.Error())
 	}
 
+	// if a body was provided ensure that the content length was set.
```

Comment:
I would have thought so however the fix does work.  I will dig into this more to understand why.